### PR TITLE
Refactor hero icon position; center on long hero sections 

### PR DESCRIPTION
**Pull request by hqhhuang**
Bug/issue #, if applicable: 93050293

## Summary
Current icon positioning looks odd on pages with longer hero sections.
This refactor centers the icons on those page, instead of pinning them to the top.

## Testing
Steps:
1. Navigate to a page that has a few items in the hero section
2. Verify that icons is centered in the hero section.
3. Navigate to a page with short hero section
4. Verify that in bigger viewports, the icon is still pinned to the top of the page. As we shrink the viewport to medium and increase the size of the sidebar, verify that the icon is dynamically positioned in the center as well.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] Added tests - NA
- [x] Ran `npm test`, and it succeeded
- [ ] Updated documentation if necessary - NA

### CODE REVIEW

**Comment by hqhhuang** [comment]
@swift-ci test

**Comment by hqhhuang** [comment]
Updated the PR with more changes to reflect the latest design: we are adding extra padding to the bottom of the hero section in IDE mode when the language switcher is shown. This is to balance the fact that, with a language switcher, visually there's more spacing on top of the hero content than below.

**Comment by hqhhuang** [comment]
@swift-ci test

**Review by dobromir-hristov — APPROVED** [review]
Looks good to me. Please add that type casting and we are good here.

**Comment by biamx3** [comment]
LGTM!

**Comment by hqhhuang** [comment]
@swift-ci test

File: src/components/DocumentationTopic.vue [hunk 1]
```diff
@@ -14,7 +14,8 @@
       <DocumentationHero
         :role="role"
         :enhanceBackground="enhanceBackground"
-        :extraPadding="extraPadding"
+        :shortHero="shortHero"
+        :shouldShowLanguageSwitcher="shouldShowLanguageSwitcher"
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply src/components/DocumentationTopic.vue:18]: @hqhhuang You have not added what I asked you here, and now I am seeing Vue warnings in CI.
```
console.error node_modules/vue/dist/vue.runtime.common.dev.js:621
      [Vue warn]: Missing required prop: "shouldShowLanguageSwitcher"
      
      found in
      
      ---> <DocumentationHero>
```

**Pull request merged**